base/float16.jl

```diff
@@ -149,8 +149,16 @@ for func in (:sin,:cos,:tan,:asin,:acos,:atan,:sinh,:cosh,:tanh,:asinh,:acosh,
         $func(a::Complex32) = complex32($func(complex64(a)))
     end
 end
-atan2(y::Float16, x::Float16) = float16(atan2(float32(y), float32(x)))
-hypot(a::Float16, b::Float16) = float16(hypot(float32(a), float32(b)))
+
+for func in (:div,:fld,:cld,:rem,:mod,:mod1,:rem1,:fld1,
```

Comment:
The `*1` functions (`mod1`, `rem1`, `fld1`) are already defined generically for all `Real` types.

(The generic versions of `rem1` and `fld1` with `Real` arguments in `operators.jl` look suspicious -- I think they are wrong; their definition is probably only correct for integers.)

Do you want to add test cases for the new functions as well?